integration-tests/integration_test.go

```diff
@@ -260,52 +260,82 @@ func (s *ProcessNetworkTestSuite) TearDownSuite() {
-	s.Require().NoError(err)
-	assert.Equal(s.T(), expectedProcessInfo, val)
+	assert := assert.New(s.T())
 
-	processName = "sleep"
-	exeFilePath = "/bin/sleep"
-	expectedProcessInfo = fmt.Sprintf("%s:%s:%d:%d", processName, exeFilePath, 0, 0)
-	val, err = s.Get(processName, processBucket)
-	s.Require().NoError(err)
-	assert.Equal(s.T(), expectedProcessInfo, val)
+	for _, expected := range expectedProcesses {
+		val, err := s.Get(expected.Name, processBucket)
+		s.Require().NoError(err)
+		processInfo, err := NewProcessInfo(val)
+		s.Require().NoError(err)
+
+		assert.Equal(expected.Name, processInfo.Name)
+		assert.Equal(expected.ExePath, processInfo.ExePath)
+		assert.Equal(expected.Uid, processInfo.Uid)
+		assert.Equal(expected.Gid, processInfo.Gid)
+		assert.True(processInfo.Pid >= 0)
+		assert.Equal(expected.Args, processInfo.Args)
+	}
 }
 
 func (s *ProcessNetworkTestSuite) TestProcessLineageInfo() {
-	processName := "awk"
-	exeFilePath := "/usr/bin/awk"
-	parentFilePath := "/bin/busybox"
-	expectedProcessLineageInfo := fmt.Sprintf("%s:%s:%s:%d:%s:%s", processName, exeFilePath, parentUIDStr, 0, parentExecFilePathStr, parentFilePath)
-	val, err := s.GetLineageInfo(processName, "0", processLineageInfoBucket)
-	s.Require().NoError(err)
-	assert.Equal(s.T(), expectedProcessLineageInfo, val)
+	expectedLineages := []ProcessLineage {
+		ProcessLineage {
+			Name: "awk",
+			ExePath: "/usr/bin/awk",
+			ParentUid: 0,
+			ParentExePath: "/bin/busybox",
+		},
+		ProcessLineage {
+			Name: "grep",
+			ExePath: "/bin/grep",
+			ParentUid: 0,
+			ParentExePath: "/bin/busybox",
+		},
+		ProcessLineage {
+			Name: "sleep",
+			ExePath: "/bin/sleep",
+			ParentUid: 0,
+			ParentExePath: "/bin/busybox",
+		},
+	}
 
-	processName = "grep"
-	exeFilePath = "/bin/grep"
-	parentFilePath = "/bin/busybox"
-	expectedProcessLineageInfo = fmt.Sprintf("%s:%s:%s:%d:%s:%s", processName, exeFilePath, parentUIDStr, 0, parentExecFilePathStr, parentFilePath)
-	val, err = s.GetLineageInfo(processName, "0", processLineageInfoBucket)
-	s.Require().NoError(err)
-	assert.Equal(s.T(), expectedProcessLineageInfo, val)
+	assert := assert.New(s.T())
 
-	processName = "sleep"
-	exeFilePath = "/bin/sleep"
-	parentFilePath = "/bin/busybox"
-	expectedProcessLineageInfo = fmt.Sprintf("%s:%s:%s:%d:%s:%s", processName, exeFilePath, parentUIDStr, 0, parentExecFilePathStr, parentFilePath)
-	val, err = s.GetLineageInfo(processName, "0", processLineageInfoBucket)
-	s.Require().NoError(err)
-	assert.Equal(s.T(), expectedProcessLineageInfo, val)
+	for _, expected := range expectedLineages {
+		val, err := s.GetLineageInfo(expected.Name, "0", processLineageInfoBucket)
+		s.Require().NoError(err)
+		lineage, err := NewProcessLineage(val)
+		s.Require().NoError(err)
+
+		assert.Equal(expected.Name, lineage.Name)
+		assert.Equal(expected.ExePath, lineage.ExePath)
+		assert.Equal(expected.ParentUid, lineage.ParentUid)
+		assert.Equal(expected.ParentExePath, lineage.ParentExePath)
```

Comment:
Think we could create a couple of helper functions that take the full maps and compares them internally? I think it would make it even easier to read. Same goes for assertion of `ProcessInfo` structs. (There's probably a way to implement comparison of structs directly, but I'm not that familiar with go 🤷🏻‍♂️ )
```suggestion
                AssertProcessLineage(expected, lineage);
```

Reply:
Looks like `assert.Equal()` might handle this for us, it will eventually do `reflect.DeepEqual()` which may or may not be suitable. I'll try it and see.